gallery/tutorials/pipeline_demo.py

```diff
@@ -0,0 +1,260 @@
+
+# with no filters
+src.projections[0:10].show()
+
+# %%
+
+# with no noise filter
+src.clean_images[0:10].show()
+
+# %%
+
+# with noise and ctf filters
+src.images[0:10].show()
+
+
+# %%
+# CTF Correction
+# --------------
+# We apply ``phase_flip()`` to correct for CTF effects.
+
+src.phase_flip()
+src.images[0:10].show()
+
+
+# %%
+# Class Averaging
+# ---------------
+# We use an ASPIRE ``RIRClass2D`` object to classify the images via the rotationally invariant
+# representation (RIR) algorithm. ``RIRClass2D`` has a ``selector`` option that selects a set
+# of images to be used as base images for classisfication. In this example, we use ``TopClassSelector``,
+# which selects the first ``n_classes`` images from the source.
+
+from aspire.classification import RIRClass2D, TopClassSelector
+
+# set parameters
+n_classes = 200
+n_nbor = 6
+
+# Create a class averaging instance
+rir = RIRClass2D(
+    src,
+    fspca_components=40,
+    bispectrum_components=30,
+    n_nbor=n_nbor,
+    n_classes=n_classes,
+    selector=TopClassSelector(),
+)
+
+# classify and average
+classes, reflections, distances = rir.classify()
+avgs = rir.averages(classes, reflections, distances)
+
+
+# %%
+# View the Class Averages
+# -----------------------
+
+# Show class averages
+avgs.images[0:10].show()
+
+# %%
+
+# Show original images corresponding to those classes
+src.images[0:10].show()
+
+
+# %%
+# Orientation Estimation
+# ----------------------
+# We initialize a ``CLSyncVoting`` class instance for estimating the orientations of the images.
+# The estimation employs the common lines method with synchronization and voting.
+
+from aspire.abinitio import CLSyncVoting
+
+# Stash true rotations for later comparison
+true_rotations = src.rotations[:n_classes]
+
+orient_est = CLSyncVoting(avgs, n_theta=72)
+
+# Get the estimated rotations
+orient_est.estimate_rotations()
+rots_est = orient_est.rotations
+
+
+# %%
+# Mean Squared Error
+# ------------------
+# ASIPRE has some built in utility functions for globally aligning the estimated rotations
+# to the true rotations and computing the mean squared error.
+
+from aspire.utils.coor_trans import (
+    get_aligned_rotations,
+    get_rots_mse,
+    register_rotations,
+)
+
+# Compare with known true rotations
+Q_mat, flag = register_rotations(rots_est, true_rotations)
+regrot = get_aligned_rotations(rots_est, Q_mat, flag)
+mse_reg = get_rots_mse(regrot, true_rotations)
+mse_reg
+
+
+# %%
+# Volume Reconstruction
+# ---------------------
+# Now that we have our class averages and rotation estimates, we can estimate the
+# mean volume by supplying the class averages and basis for back projection.
+
+from aspire.basis import FFBBasis3D
+from aspire.reconstruction import MeanEstimator
+
+# Assign the estimated rotations to the class averages
+avgs.rotations = rots_est
+
+# Create a reasonable Basis for the 3d Volume
+basis = FFBBasis3D((res,) * 3, dtype=vol.dtype)
```

Comment:
Should be able to just instantiate with `res` alone, looks cleaner

Reply:
Ah yes, looks better. Done.